### PR TITLE
fix: /api/nodes is timeout when emqx in high load

### PR DESCRIPTION
**Pull request by zhongwencool**
1. Fix https://github.com/emqx/emqx/issues/9353
When OS resources are limited (CPU utilization > 90%), it causes the node API to be slow (> 20 seconds), and the dashboard to report errors continuously.

After caching erlang version and adding a timeout to fetch sys_memory,
this API only takes 40~50ms even when CPU utilization > 95%.

2. Ensure that the precision of the alarm's usage is 2, and add more test cases.
3. Correct the type of `emqx_vm:loads().` to float, not string.

https://emqx.atlassian.net/browse/EMQX-8801

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_sys_memory.erl [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+init([]) ->
+    {ok, #{last_time => 0}}.
+
+handle_call(get_sys_memory, _From, State = #{last_time := LastTime}) ->
+    Now = erlang:system_time(millisecond),
+    case Now - LastTime >= ?TIMEOUT of
+        true ->
+            Memory = load_ctl:get_sys_memory(),
+            persistent_term:put(?SYS_MEMORY_CACHE_KEY, Memory),
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_sys_memory.erl:54]: Maybe don't use persisten_term for frequently updated cache value.
We can use ETS for this.

**Comment by zhongwencool** [comment]
> > After caching erlang version and adding a timeout to fetch sys_memory,
> 
> How often do we fetch sys_memory? I am just afraid to return a cached value may mislead the user about what the real problem is.

Request every 2 seconds. 
In my opinion, it's better to return a cache value than to keep time_out.
but I can change to use cache value only when gen_server call timeout(2s), otherwise, use the real value.

**Comment by qzhuyan** [comment]
We could have some timeout and return error or use `0` to indicate timeout but shall not return the cached value.
Reading system memory timeout indicates a serious resource issue, using cached value hides the problem.

During the performance troubleshooting the failed/timeout of memory usage reading combined with CPU usage could help us locates where the constraints are.

File: apps/emqx_management/src/emqx_mgmt_sys_memory.erl (hunk 1 above)
Comment (zhongwencool) [reply apps/emqx_management/src/emqx_mgmt_sys_memory.erl]: DONE

File: apps/emqx_management/src/emqx_mgmt_cache.erl [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+get_sys_memory() ->
+    Now = now_millisecond(),
+    {CacheMem, ExpiredAt} = get_memory_from_cache(),
+    case Now > ExpiredAt of
+        true ->
+            erlang:send(?MODULE, fresh_sys_memory),
+            CacheMem;
+        %% stale cache value, try to recalculate
+        false ->
+            get_sys_memory_sync()
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cache.erl:38]:
```suggestion
    case Now < ExpiredAt of
        true ->
            erlang:send(?MODULE, fresh_sys_memory),
            CacheMem;
        %% stale cache value, try to recalculate
        false ->
            get_sys_memory_sync()
```

File: apps/emqx_management/src/emqx_mgmt_cache.erl [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+-define(REFRESH_MS, ?EXPIRED_MS - 100).
+-define(DEFAULT_BAD_MEMORY, {0, 0}).
+
+-export([start_link/0, get_sys_memory/0]).
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2, code_change/3]).
+
+get_sys_memory() ->
+    Now = now_millisecond(),
+    {CacheMem, ExpiredAt} = get_memory_from_cache(),
+    case Now > ExpiredAt of
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cache.erl:32]: maybe add a `is_expired` help function, or make get_memory_from_cache/1 return `{ok, CacheMem}`, or `stale`

Comment (zhongwencool) [reply apps/emqx_management/src/emqx_mgmt_cache.erl]: good idea

File: apps/emqx_management/src/emqx_mgmt_cache.erl [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    ok.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+fresh_sys_memory(State = #{fresh_at := LastFreshAt}) ->
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cache.erl:78]: nit:
```suggestion
refresh_sys_memory(State = #{last_refresh := LastRefresh}) ->
```

**Review by qzhuyan — APPROVED** [review]

File: apps/emqx/src/emqx_os_mon.erl [hunk 5]
```diff
@@ -236,5 +244,5 @@ do_update_mem_alarm_status(HWM0) ->
     ok.
 
 usage_msg(Usage, What) ->
-    %% devide by 1.0 to ensure float point number
+    %% divide by 1.0 to ensure float point number
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_os_mon.erl:247]: nit, should `float(Int)` help here?

File: apps/emqx_management/src/emqx_mgmt_cache.erl [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+init([]) ->
+    ets:new(?MODULE, [set, named_table, public, {keypos, 1}]),
+    {ok, #{fresh_at => 0}}.
+
+handle_call(get_sys_memory, _From, State) ->
+    {Mem, NewState} = fresh_sys_memory(State),
+    {reply, Mem, NewState};
+handle_call(_Request, _From, State) ->
+    {reply, ok, State}.
```
Comment (qzhuyan) [inline apps/emqx_management/src/emqx_mgmt_cache.erl:57]: nit, better return something other than `ok`

**Pull request merged**